projects/pre-sentence-reports-to-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/integrations/delius/document/DocumentService.kt

```diff
@@ -0,0 +1,72 @@
+package uk.gov.justice.digital.hmpps.integrations.delius.document
+
+import org.springframework.http.HttpEntity
+import org.springframework.http.MediaType
+import org.springframework.http.client.MultipartBodyBuilder
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+import org.springframework.util.MultiValueMap
+import uk.gov.justice.digital.hmpps.audit.service.AuditableService
+import uk.gov.justice.digital.hmpps.audit.service.AuditedInteractionService
+import uk.gov.justice.digital.hmpps.config.ServiceContext
+import uk.gov.justice.digital.hmpps.exception.ConflictException
+import uk.gov.justice.digital.hmpps.exception.NotFoundException
+import uk.gov.justice.digital.hmpps.integrations.alfresco.AlfrescoClient
+import uk.gov.justice.digital.hmpps.integrations.delius.audit.BusinessInteractionCode
+import uk.gov.justice.digital.hmpps.integrations.delius.courtreport.CourtReportRepository
+import uk.gov.justice.digital.hmpps.message.AdditionalInformation
+import uk.gov.justice.digital.hmpps.message.SimpleHmppsEvent
+import java.net.URLEncoder
+import java.nio.charset.StandardCharsets
+
+@Service
+class DocumentService(
+    auditedInteractionService: AuditedInteractionService,
+    private val documentRepository: DocumentRepository,
+    private val courtReportRepository: CourtReportRepository,
+    private val alfrescoClient: AlfrescoClient
+) : AuditableService(auditedInteractionService) {
+    fun AdditionalInformation.reportId() = this["reportId"] as String
+    @Transactional
+    fun updateCourtReportDocument(hmppsEvent: SimpleHmppsEvent, file: ByteArray) =
+        audit(BusinessInteractionCode.UPLOAD_DOCUMENT) {
+            val reportId = hmppsEvent.additionalInformation.reportId()
```

Comment:
The `reportId` will be a UUID, but we need a URN for looking up by external reference.  I _think_ the JSON response from the detailUrl gives us the URN we need.